### DISCUSSION TITLE
Why write quorum should be the number of data drives + 1 where code rate=1/2

### DISCUSSION THREAD

**Discussion by parksworks** [minio/minio#17038] [Q&A] [answered]
When code rate >1/2, in minio, the write quorum is the number of data drives(blocks).
It looks like a natural constraint.

But for the case of code rate =1/2, the write quorum is the number of data drives +1.
In this case, why +1 is applied to the write quorum?
If it required to check the consistency of the coded data(with 1 parity), then why the write quorum for the case of code rate >1/2 can be the same as the number of data drives?

**Comment by harshavardhana** [comment, marked as answer]
For split brain, which is real in the real world.

**Reply by parksworks** [reply]
Could you give me more details about the rate >1/2 case that exploits only the data drives? 
As far as I know, the split brain problem occur in data replication configuration, but I want to use only erasure coding without replication.
In my configuration example, for each 4 servers equipped 4 drives with 12+4 erasure coding, the write quorum be 12 and write is possible where 1 server is unavailable.
In this case, we dont have any chance to check correctness of the stored data.
Could you explain how the split brain problem can occur and the 12 alive drives can prevent the problem?

**Reply by harshavardhana** [reply]
Split brain is not just for replication, it's meant for any quorum state. A state can never use write quorum to be n/2 when number of data shards are same as parity shards. 

Write always guarantees that there should be no inconsistent state() if the write has succeeded.

To make sure that there are no inconsistent states()

One must ensure that you never write inconsistently.
